### PR TITLE
[WIP] Diagnosing Vulkan VMA removal issue #14776.

### PR DESCRIPTION
**Pull request by benvanik**
Progress on #13945.

ci-extra: test_a100

### CODE REVIEW

File: runtime/src/iree/hal/drivers/vulkan/registration/driver_module.cc [hunk 1]
```diff
@@ -42,7 +42,7 @@ IREE_FLAG(bool, vulkan_sparse_residency, true,
 IREE_FLAG(
     bool, vulkan_dedicated_compute_queue, false,
     "Use a dedicated queue with VK_QUEUE_COMPUTE_BIT for dispatch workloads.");
-IREE_FLAG(bool, vulkan_vma_allocator, true,
+IREE_FLAG(bool, vulkan_vma_allocator, false,
```
Comment (ScottTodd) [inline runtime/src/iree/hal/drivers/vulkan/registration/driver_module.cc]: Interesting test failures... I wonder if they reproduce consistently.
https://github.com/openxla/iree/actions/runs/5927315254/job/16070945531?pr=14760#step:7:833
`/work/runtime/src/iree/hal/drivers/vulkan/native_allocator.cc:300: RESOURCE_EXHAUSTED; VK_ERROR_OUT_OF_DEVICE_MEMORY; vkAllocateMemory; failed to allocate buffer of length 24; while invoking native function hal.allocator.allocate; while calling import; `

Comment (benvanik) [reply runtime/src/iree/hal/drivers/vulkan/registration/driver_module.cc]: yeah very weird - building now to try to reproduce locally

Comment (ScottTodd) [reply runtime/src/iree/hal/drivers/vulkan/registration/driver_module.cc]: That job is running with `--env CTEST_PARALLEL_LEVEL=2`: https://github.com/openxla/iree/blob/617a1d8fc39d22af233616b406c0f6f85568eef6/.github/workflows/ci.yml#L314

So unlikely that multiple tests _running at the same time_ are exhausting device memory. If the individual tests aren't the issue, maybe device memory isn't being freed / reclaimed?

Comment (benvanik) [reply runtime/src/iree/hal/drivers/vulkan/registration/driver_module.cc]: Hrm these work just fine for me locally :/

**Comment by benvanik** [comment]
Closing because @ScottTodd's workaround in #14790 seems good enough for now.

**Pull request closed without merging**